etcdserver/apply.go

```diff
@@ -375,15 +383,22 @@ func (a *applierV3backend) Txn(rt *pb.TxnRequest) (*pb.TxnResponse, error) {
 	// So the reader will not see any intermediate results.
 	txnID := a.s.KV().TxnBegin()
 
+	var txerr error
 	resps := make([]*pb.ResponseOp, len(reqs))
 	for i := range reqs {
-		resps[i] = a.applyUnion(txnID, reqs[i])
+		resps[i], txerr = a.applyUnion(txnID, reqs[i])
```

Comment:
there needs to be a `checkRequestPut` with the other check functions, otherwise this could partially apply the txn by doing a Put then erroring out on the next Put